hockeysdk/src/main/java/net/hockeyapp/android/CrashManager.java

```diff
@@ -403,64 +403,72 @@ public static synchronized void submitStackTraces(final WeakReference<Context> w
     }
 
     private static void submitStackTrace(final WeakReference<Context> weakContext, String filename, CrashManagerListener listener, CrashMetaData crashMetaData) {
+        String stacktrace = null;
+        try {
+            stacktrace = contentsOfFile(weakContext, filename);
+        } catch (Exception e) {
+            HockeyLog.error("Failed to read crash data", e);
```

Comment:
If this catch block is executed, then we go to line 412 and check length on a null stacktrace object. Shouldn't that crash? According to me the new unit test covers only empty stack trace case. I would suggest calling listener and cleaning retry counter too for null case.